app/controllers/api/diagnosed_needs_controller.rb

```diff
@@ -3,23 +3,30 @@
 module Api
   class DiagnosedNeedsController < ApplicationController
     def index
-      @diagnosed_needs = DiagnosedNeed.where(diagnosis_id: params[:diagnosis_id])
+      diagnosis = Diagnosis.find params[:diagnosis_id]
+      not_found unless diagnosis.can_be_viewed_by?(current_user)
+      @diagnosed_needs = DiagnosedNeed.where(diagnosis: diagnosis)
     end
 
     def bulk
-      DiagnosedNeed.transaction do
-        DiagnosedNeed.create bulk_create_param_array
-        DiagnosedNeed.update(bulk_update_param_hash.keys, bulk_update_param_hash.values)
-        DiagnosedNeed.destroy bulk_delete_param_array
-      end
-
+      diagnosis = Diagnosis.find params[:diagnosis_id]
+      not_found unless diagnosis.can_be_viewed_by?(current_user)
```

Comment:
Why is this one not in the same helper function style as the other ? 
Like `check_access_to_visit(visit)`